### PR TITLE
Folder structure for code snippets

### LINKED ISSUE #6131: Add organizational folders to snippets
All snippets currently go in a folder called "all" for a historical reason I don't full understand.  https://github.com/rerun-io/rerun/pull/6120 starts to introduce view snippets, which is only going to make the already-overhwhelming structure moreso.

We should introduce a structure to organize these into:
 - archetypes/
 - views/
 - tutorials/

etc.

### LINKED ISSUE #6132: Add organizational folders to snippets

### LINKED ISSUE #6133: All folder structure to snippets
All snippets currently go in a folder called "all" for a historical reason I don't full understand.  https://github.com/rerun-io/rerun/pull/6120 starts to introduce view snippets, which is only going to make the already-overhwhelming structure moreso.

We should introduce a structure to organize these into:
 - archetypes/
 - views/
 - tutorials/

etc.

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #6131
* Fixes #6132
* Fixes #6133

(either someone fell victim to a github glitch, or really liked creating issues about this 😄)

The compare snippet script needed quite a bit of love to make this work.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6219?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6219?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6219)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by Wumpf** [comment]
lots of things failing, including webpage build

**Comment by Wumpf** [comment]
@rerun-bot full-check

File: crates/re_viewer/data/quick_start_guides/quick_start_spawn.cpp
Comment (jleibs) [inline crates/re_viewer/data/quick_start_guides/quick_start_spawn.cpp:1]: It looks like this used to be a symlink which means it was linking to a thing that was threotically being round-trip-tested. I'm not seeing a mechanism that causes that to be the case now.

Comment (jleibs) [reply crates/re_viewer/data/quick_start_guides/quick_start_spawn.cpp:1]: Actually, does anything even use this anymore? Didn't we get rid of embedded quick starts in the viewer?

**Review by jleibs — APPROVED** [review]

**Comment by Wumpf** [comment]
had to do some bigger changes: the toml wasn't valid for ` pixi run -e wheel-test build-examples snippets example_data/snippets`, see https://github.com/rerun-io/rerun/actions/runs/9005576258/job/24741561218
Turned around a bit and removed support for magic  "example" named files and support arbitrary subdirs more consistently for the most part. But didn't get things to pass all tests locally on windows. Direction should be correct

**Comment by Wumpf** [comment]
(do not merge until main ci is green with this)

**Comment by jleibs** [comment]
@rerun-bot full-check

**Pull request merged**